src/lib/Scenes/MyCollection/Screens/ArtworkForm/MyCollectionArtworkForm.tsx

```diff
@@ -169,7 +169,10 @@ export const MyCollectionArtworkForm: React.FC<MyCollectionArtworkFormProps> = (
 
   const onHeaderBackButtonPress = () => {
     const currentRoute = navContainerRef.current?.getCurrentRoute()
-    const isFirstScreen = props.mode === "edit" || !currentRoute?.name || currentRoute?.name === "ArtworkFormArtist"
+    const isFirstScreen =
+      (props.mode === "edit" && currentRoute?.name !== "AddPhotos") ||
```

Comment:
Do we need to check for `mode === "edit"` here or would it be the same for "create"? 🤔 

```suggestion
      currentRoute?.name !== "AddPhotos" ||
```

Reply:
I believe that we need to, because when I used the suggested change and tested adding a new artwork, after searching for an artist and choosing artists then choosing an artwork, tapping the back button takes me to `My Collection`, instead of the selected artist's artworks

Reply:
Alright, then let's keep it as it is 👍

Reply:
there might be an other way around to do this by simply using navigation.goBack() inside google photos instead of relying on `onHeaderBackButtonPress`. I will make a PR for it